src/runtimeHelpers.ts

```diff
@@ -253,6 +297,109 @@ export function unPartitionResults<T>(
         });
 }
 
+/**
+ * Perform the inverse mapping from partitionItems on the nested results we get
+ * back from the service. This function is only called when we have secondaryBatchKey.
+ * We assume the newKey is inside the response here.
+ *
+ * Example
+ * ```js
+ * unPartitionResultsByBatchKeyPartition(
+ *   'bard_id',
+ *   [ ['name', 'rating'], ['rating'] ],
+ *   [ [2, 2], [1] ],
+ *   [ [0, 2], [1] ],
+ *   [
+ *      [ { bar_id: 2, name: 'Burger King', rating: 3 } ],
+ *      [ { bar_id: 1, name: 'In N Out', rating: 4 }  ]
+ *   ],
+ * )
+ * ```
+ *
+ * Returns:
+ * ```
+ * [
+ *   { bar_id: 2, name: 'Burger King' }，
+ *   { bar_id: 1, rating: 4 }，
+ *   { bar_id: 2, rating: 3 }，
+ * ]
+ */
+export function unPartitionResultsByBatchKeyPartition<T>(
+    newKey: string,
+    batchKeyPartition: ReadonlyArray<ReadonlyArray<any>>,
+    secondaryBatchKeyPartion: ReadonlyArray<ReadonlyArray<any>>,
+    /** Should be a nested array of IDs, as generated by partitionItems */
+    requestGroups: ReadonlyArray<ReadonlyArray<number>>,
+    /** The results back from the service, in the same shape as groups */
+    resultGroups: ReadonlyArray<ReadonlyArray<T | CaughtResourceError | BatchItemNotFoundError>>,
+): ReadonlyArray<T | Error> {
+    /**
+     * e.g. with our inputs, produce:
+     * ```js
+     * [
+     *   [
+     *      { order: 0, result: { bar_id: 2, name: 'Burger King' },
+     *      { order: 2, result: { bar_id: 2, rating: 3 } },
+     *   ],
+     *   [
+     *      { order: 1, result: { bar_id: 1, rating: 4 } },
+     *   ]
+     * ]
+     * ```
+     */
+    const zippedGroups = requestGroups.map(function (ids, i) {
```

Comment:
```suggestion
    const zippedGroups = requestGroups.map((ids, i) => {
```

prefer using arrow syntax where possible to avoid binding a new `this`